test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -37,7 +37,7 @@ public void AutoFitBinaryTest()
         [Fact]
         public void AutoFitMultiTest()
         {
-            var context = new MLContext(1);
+            var context = new MLContext(42);
```

Comment:
@justinormont before I changed the seed in `EnsureGroupPreservationColum` to fall back to the `MLContext` seed, the `GenerateNumverTransform` was using its default `seed=42`. With the change, when it defaulted to `seed=1` from `MLContext`, the cross validation microaccuracy for this test fell from ~0.71 to ~0.57. Changing the `MLContext` seed to 42 gets the micro accuracy back to ~0.57. Very curious to see so much delta in micro accuracy based solely on the data splits.

Reply:
`42` is a great default seed :)

The [dataset ](https://github.com/dotnet/machinelearning/blob/553e089259b228c29403fd71bd006d7fdef3e0db/test/Microsoft.ML.AutoML.Tests/TestData/TrivialMulticlassDataset.txt) being 180 rows, I would expect swings in a model accuracy. Though I would expect our use of 10-fold cross-validation to keep the value pretty stable, even for small datasets.

Can you check that cross-validation still gets independent folds (and doesn't keep recreating the exact same fold due to the fixed seed)? We may need to modify the CV fold creation, perhaps by using `seed = seed + foldNum`.

Reply:
> Changing the `MLContext` seed to 42 gets the micro accuracy back to ~0.71. 

If changing the seed back to 42 fixes the accuracy, it's likely not the folds being recreated; though still useful to check. 

Perhaps trying a few seed values will tell us the normal range for the model accuracy. 

Side note: Technically the AutoML returns the metrics of the fold [closest to the average](https://github.com/dotnet/machinelearning/blob/e50c4d20012e0d62852f404ae443afca7dad043e/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L75-L78) for CV instead of the average, which is a bug we should fix. Since AutoML will automatically choose CV or TrainTest and can have only one return type, and AutoML couldn't at the time create a metric with its own values, CV also returned the singular metric class instead of the CV metric stats class. Should switch to always return `MetricStatistics` instance.

Reply:
Yes, the folds are independent. Changing the seed only affects how the `SamplingKeyColumn` in the `IDataView data` coming into this method is randomized. The folds are then created in the same way as before here:
https://github.com/dotnet/machinelearning/blob/46e7dc6269ef4a55c0e5788f0c327ef2029ee2f2/src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs#L465-L473

Reply:
Thanks @najeeb-kazmi for looking, and you're completely correct.